.github/workflows/phpstan.yml

```diff
@@ -0,0 +1,65 @@
+name: 'PHPStan Analysis'
+
+on:
+    pull_request:
+        paths:
+            - 'plugins/woocommerce/**/*.php'
+            - 'plugins/woocommerce/composer.json'
+            - 'plugins/woocommerce/composer.lock'
+            - 'plugins/woocommerce/phpstan.neon'
+            - 'plugins/woocommerce/phpstan-baseline.neon'
+    push:
+        paths:
+            - 'plugins/woocommerce/**/*.php'
+            - 'plugins/woocommerce/composer.json'
+            - 'plugins/woocommerce/composer.lock'
+            - 'plugins/woocommerce/phpstan.neon'
+            - 'plugins/woocommerce/phpstan-baseline.neon'
+        branches:
+            - 'trunk'
+            - 'release/*'
+
+concurrency:
+    group: phpstan-${{ github.event_name == 'push' && github.run_id || github.event_name }}-${{ github.ref }}
+    cancel-in-progress: true
+
+env:
+    FORCE_COLOR: 1
+
+jobs:
+    phpstan:
+        name: 'PHPStan Analysis'
+        runs-on: ubuntu-latest
+        steps:
+            - uses: 'actions/checkout@v4'
+              name: 'Checkout'
+
+            - uses: './.github/actions/setup-woocommerce-monorepo'
+              id: 'setup-monorepo'
+              with:
+                  install: '@woocommerce/plugin-woocommerce...'
+                  pull-package-deps: '@woocommerce/plugin-woocommerce'
+                  php-version: '8.4'
```

Comment:
Same as [other comment](https://github.com/woocommerce/woocommerce/pull/62064/files#r2555801077), should this be run on a lower PHP version?